---
title: "RNA-seq data analysis training"
output:
  html_document:
    df_print: paged
    toc: true
    toc_depth: 5
---

---

- Guillem Ylla, Ph.D.
- Harvard University
- September 2021

Data obtained from:

- Tarikere, S., Ylla, G.,& Extavour, C. G. "Distinct gene expression dynamics in germ line and somatic tissue during ovariole morphogenesis in *Drosophila melanogaster*", 2021.
- Github repo of the original data and analysis: https://htmlpreview.github.io/?https://raw.githubusercontent.com/guillemylla/Ovariole_morphogenesis_RNAseq/

---



## Load libraries

```{r message=FALSE, warning=FALSE}
library(ggplot2)
library(tidyr)
library(dplyr)
library(DESeq2)

knitr::opts_chunk$set(fig.width = 12)
knitr::opts_chunk$set(fig.height =  8)
knitr::opts_chunk$set(cache = TRUE)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(warnings = FALSE)
```


## Load Data

- Load counts table
  - RNA-seq mapping was performed with RSEM in this [script](Run_RSEM_batch.sh)


```{r,warning=FALSE,results=FALSE}
Metadata<-read.csv("data/Metadata.csv")
rownames(Metadata)<-Metadata$BioSample#set BioSample as rownames 
Metadata
```


```{r,warning=FALSE,results=FALSE}
Counts<-read.csv(file="data/Counts_table.csv", row.names = 1) 
head(Counts)
dim(Counts)
```




## Normalitzation 


- Remove genes with ZERO counts in all samples

```{r}
Counts_nz<-Counts[rowSums(Counts)!=0,]

dim(Counts_nz)
```

- Round to the unit because DESeq2 wants integer counts.

```{r}
Counts_nz<-round(Counts_nz,0)
```

- Make sure that the row names of the metadata file are in the same than the table of counts column name

```{r,warning=FALSE,results=FALSE}
identical(colnames(Counts_nz), rownames(Metadata))
```

- Load data in DESeq2. As "design" indicate the factor that we want to compare.

```{r,warning=FALSE,results=FALSE}
dds <- DESeq2::DESeqDataSetFromMatrix(countData = Counts_nz,
                              colData = Metadata,
                              design= ~Stage)
#estimate factors
dds <- DESeq2::DESeq(dds)
```

- Get the VST normalized reads in case you want them for something (i.e. PCA)

```{r}
VST <- assay(varianceStabilizingTransformation(dds, blind=TRUE))
head(VST)
```

##  Differential Expression Analysis

- Run DEA

```{r}
dea_result <- DESeq2::DESeq(dds)
```

- See top DEGs

```{r}
dea_result_df<-as.data.frame(results(dea_result, contrast = c("Stage", "Early","Mid"), alpha=0.001) )

head(dea_result_df)

```


- Count DEGs

```{r}
dea_result_df %>% 
  filter(padj <0.01 ) %>% 
  dplyr::summarise(sum(log2FoldChange<0),
            sum(log2FoldChange>0)) 
```

## Graphic visualiaztion

I like to use ggplots and other Tidyverse packages, for which a long data format is better.


- Transform VST table to long foramt and add metadata

```{r}
VST_meta<-as.data.frame(VST) %>% 
  tibble::rownames_to_column(var = "GeneID") %>% 
  pivot_longer(cols=colnames(VST),names_to="Sample", values_to="VST") %>% 
  left_join(Metadata, by=c("Sample"="BioSample")) 

head(VST_meta)
```


- Boxplot normalized data

```{r}
Boxplot_VST <- ggplot(VST_meta, aes(x=Sample, y=VST,  fill=Stage)) +
  geom_boxplot()+ 
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, hjust = .8, size=8))+
  labs(title = "Normalized Counts per gene (VST)",
              subtitle = "Normalized counts per gene in each sample. Only 3 replicates per stage",fill ="Legend")+xlab("")

Boxplot_VST
```

- PCA

```{r}
## Prepare data for PCA
VST_toPCA<-t(VST) # transpose data (is how prcomp wants it)

## add metadata
VST_toPCA<-as.data.frame(VST_toPCA) %>%  
    tibble::rownames_to_column(var ="BioSample") %>% 
    dplyr::left_join(Metadata, by=c("BioSample"="BioSample"))

rownames(VST_toPCA)<-VST_toPCA$BioSample

#compute PCA
pca_VST<-prcomp(VST_toPCA[,2:nrow(VST)+1], scale. =FALSE  )

#get variance
var_explained <- pca_VST$sdev^2/sum(pca_VST$sdev^2)

#Plot PCA with ggplots
pca_VST_plot<-pca_VST$x %>% 
  as.data.frame %>%
  ggplot(aes(x=PC1,y=PC2))  +
  geom_point(aes(color=VST_toPCA$Stage), size = 8)+
  labs(x=paste0("PC1: ",round(var_explained[1]*100,1),"%"),
       y=paste0("PC2: ",round(var_explained[2]*100,1),"%"),
       title = "PCA all genes VST",color ="Color", shape="Shape")+
  theme_bw(base_size = 12)

pca_VST_plot
```

- Volcano plot of DEA


```{r}
ggplot(dea_result_df, aes(x=log2FoldChange, y=-log10(padj)) ) +
  geom_point(data = subset(dea_result_df, padj < 0.01 & abs(log2FoldChange)>2),   color = "red", alpha = 0.8)+
  geom_point(data = subset(dea_result_df, padj > 0.01 |  abs(log2FoldChange)<2 ),   color = "black", alpha = 0.8) +
  geom_hline(yintercept = 2, linetype = "dashed", color="blue")+
  geom_vline(xintercept = c(-2, 2), linetype = "dashed", color="blue") 

```

