<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Strict//EN">
<html>
<head><title>MIII_27_GGCTAC.R1.fastq.gz FastQC Report</title>

<style type="text/css">

 @media screen {
  div.summary {
    width: 18em;
    position:fixed;
    top: 3em;
    margin:1em 0 0 1em;
  }
  
  div.main {
    display:block;
    position:absolute;
    overflow:auto;
    height:auto;
    width:auto;
    top:4.5em;
    bottom:2.3em;
    left:18em;
    right:0;
    border-left: 1px solid #CCC;
    padding:0 0 0 1em;
    background-color: white;
    z-index:1;
  }
  
  div.header {
    background-color: #EEE;
    border:0;
    margin:0;
    padding: 0.5em;
    font-size: 200%;
    font-weight: bold;
    position:fixed;
    width:100%;
    top:0;
    left:0;
    z-index:2;
  }

  div.footer {
    background-color: #EEE;
    border:0;
    margin:0;
	padding:0.5em;
    height: 1.3em;
	overflow:hidden;
    font-size: 100%;
    font-weight: bold;
    position:fixed;
    bottom:0;
    width:100%;
    z-index:2;
  }
  
  img.indented {
    margin-left: 3em;
  }
 }
 
 @media print {
	img {
		max-width:100% !important;
		page-break-inside: avoid;
	}
	h2, h3 {
		page-break-after: avoid;
	}
	div.header {
      background-color: #FFF;
    }
	
 }
 
 body {    
  font-family: sans-serif;   
  color: #000;   
  background-color: #FFF;
  border: 0;
  margin: 0;
  padding: 0;
  }
  
  div.header {
  border:0;
  margin:0;
  padding: 0.5em;
  font-size: 200%;
  font-weight: bold;
  width:100%;
  }    
  
  #header_title {
  display:inline-block;
  float:left;
  clear:left;
  }
  #header_filename {
  display:inline-block;
  float:right;
  clear:right;
  font-size: 50%;
  margin-right:2em;
  text-align: right;
  }

  div.header h3 {
  font-size: 50%;
  margin-bottom: 0;
  }
  
  div.summary ul {
  padding-left:0;
  list-style-type:none;
  }
  
  div.summary ul li img {
  margin-bottom:-0.5em;
  margin-top:0.5em;
  }
	  
  div.main {
  background-color: white;
  }
      
  div.module {
  padding-bottom:1.5em;
  padding-top:1.5em;
  }
	  
  div.footer {
  background-color: #EEE;
  border:0;
  margin:0;
  padding: 0.5em;
  font-size: 100%;
  font-weight: bold;
  width:100%;
  }


  a {
  color: #000080;
  }

  a:hover {
  color: #800000;
  }
      
  h2 {
  color: #800000;
  padding-bottom: 0;
  margin-bottom: 0;
  clear:left;
  }

  table { 
  margin-left: 3em;
  text-align: center;
  }
  
  th { 
  text-align: center;
  background-color: #000080;
  color: #FFF;
  padding: 0.4em;
  }      
  
  td { 
  font-family: monospace; 
  text-align: left;
  background-color: #EEE;
  color: #000;
  padding: 0.4em;
  }

  img {
  padding-top: 0;
  margin-top: 0;
  border-top: 0;
  }

  
  p {
  padding-top: 0;
  margin-top: 0;
  }
  
</style>

</head>
<body>
<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Tue 12 Dec 2017<br />
MIII_27_GGCTAC.R1.fastq.gz
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>MIII_27_GGCTAC.R1.fastq.gz</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>23347348</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>50</td>
</tr>
<tr>
<td>%GC</td>
<td>49</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/error.png" alt="[FAIL]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>ATTGCGACCTCGATGTTGGATTAAGATATAATTTTGGGTGTAGCCGTTCA</td>
<td>73447</td>
<td>0.31458390905896466</td>
<td>No Hit</td>
</tr>
<tr>
<td>AAAGATTGCGACCTCGATGTTGGATTAAGATATAATTTTGGGTGTAGCCG</td>
<td>48309</td>
<td>0.20691429279248333</td>
<td>No Hit</td>
</tr>
<tr>
<td>TTTTGGAGAGTTCATATCGATAAAAAAGATTGCGACCTCGATGTTGGATT</td>
<td>44027</td>
<td>0.18857387999699152</td>
<td>No Hit</td>
</tr>
<tr>
<td>TTTGGGTGTAGCCGTTCAAATTTTAAGTCTGTTCGACTTTTAAATTCTTA</td>
<td>41622</td>
<td>0.17827292418822044</td>
<td>No Hit</td>
</tr>
<tr>
<td>TATCACAACGACAAGTAATTTACCCAATTCAACTAAATTCATCAATTGAA</td>
<td>38581</td>
<td>0.1652478902528887</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGTGAAGATCAAGAACTTCGGCATCTGGCTGCGTTACGATTCCCGCTCGG</td>
<td>37783</td>
<td>0.16182994316956256</td>
<td>No Hit</td>
</tr>
<tr>
<td>CGTTTCGTTCGCTGGCCCAAATACATCCGGGTGCAGCGTCAAAAGGCTGT</td>
<td>35511</td>
<td>0.15209864520801247</td>
<td>No Hit</td>
</tr>
<tr>
<td>TCGGACAAGGCTCAGGATGCCGTTGTGAAGGTCTTCATTGGACCCAAGTA</td>
<td>29699</td>
<td>0.12720502559862473</td>
<td>No Hit</td>
</tr>
<tr>
<td>TTGGACTCGCTCGTACTACAACTGGTGCCCGTGTGTTTGGCGCTATGAAG</td>
<td>26984</td>
<td>0.11557629586024074</td>
<td>No Hit</td>
</tr>
<tr>
<td>AATGCTGAAGATCTTTTCGAGAATATGAAATAATAAATTCGTCTACAATT</td>
<td>25805</td>
<td>0.11052647178600328</td>
<td>No Hit</td>
</tr>
<tr>
<td>TTTGGAGAGTTCATATCGATAAAAAAGATTGCGACCTCGATGTTGGATTA</td>
<td>24538</td>
<td>0.10509973124142409</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>AAAAA</td>
<td>7204200</td>
<td>4.2874804</td>
<td>7.604898</td>
<td>45</td>
</tr>
<tr>
<td>TGGAC</td>
<td>2662570</td>
<td>2.5657256</td>
<td>9.746723</td>
<td>1</td>
</tr>
<tr>
<td>AAGAA</td>
<td>3635255</td>
<td>2.3842082</td>
<td>5.259945</td>
<td>1</td>
</tr>
<tr>
<td>GCTGG</td>
<td>2226305</td>
<td>2.3642135</td>
<td>5.0761538</td>
<td>11</td>
</tr>
<tr>
<td>ATTTT</td>
<td>1958720</td>
<td>2.3091583</td>
<td>6.7162976</td>
<td>31</td>
</tr>
<tr>
<td>AAGGA</td>
<td>2936240</td>
<td>2.1222358</td>
<td>5.6506</td>
<td>1</td>
</tr>
<tr>
<td>TTGGA</td>
<td>2028945</td>
<td>2.063977</td>
<td>5.20252</td>
<td>16</td>
</tr>
<tr>
<td>TCGAT</td>
<td>1891750</td>
<td>1.9624397</td>
<td>6.1902776</td>
<td>10</td>
</tr>
<tr>
<td>TTTGG</td>
<td>1594165</td>
<td>1.9239072</td>
<td>7.58851</td>
<td>1</td>
</tr>
<tr>
<td>CGATG</td>
<td>1982780</td>
<td>1.9106613</td>
<td>5.762005</td>
<td>11</td>
</tr>
<tr>
<td>TTTTG</td>
<td>1460150</td>
<td>1.8970191</td>
<td>6.95722</td>
<td>32</td>
</tr>
<tr>
<td>GGACA</td>
<td>2278435</td>
<td>1.8506725</td>
<td>6.380502</td>
<td>2</td>
</tr>
<tr>
<td>AGGAC</td>
<td>2254090</td>
<td>1.830898</td>
<td>6.2098203</td>
<td>1</td>
</tr>
<tr>
<td>TGGAT</td>
<td>1793445</td>
<td>1.8244109</td>
<td>5.30804</td>
<td>17</td>
</tr>
<tr>
<td>AATTT</td>
<td>1787415</td>
<td>1.7761947</td>
<td>5.2578716</td>
<td>30</td>
</tr>
<tr>
<td>GATGT</td>
<td>1584910</td>
<td>1.6122754</td>
<td>5.108254</td>
<td>12</td>
</tr>
<tr>
<td>TGTTG</td>
<td>1333910</td>
<td>1.6098201</td>
<td>5.8359456</td>
<td>14</td>
</tr>
<tr>
<td>GGACG</td>
<td>1728710</td>
<td>1.547419</td>
<td>5.6438084</td>
<td>2</td>
</tr>
<tr>
<td>ATTGC</td>
<td>1464695</td>
<td>1.5194268</td>
<td>6.0311832</td>
<td>1</td>
</tr>
<tr>
<td>GTTGG</td>
<td>1275395</td>
<td>1.4297895</td>
<td>5.211208</td>
<td>15</td>
</tr>
<tr>
<td>TAATT</td>
<td>1385645</td>
<td>1.3769466</td>
<td>5.147251</td>
<td>29</td>
</tr>
<tr>
<td>TCGCT</td>
<td>1170245</td>
<td>1.3642696</td>
<td>5.1411324</td>
<td>9</td>
</tr>
<tr>
<td>ATGTT</td>
<td>1224875</td>
<td>1.3413736</td>
<td>5.0069923</td>
<td>13</td>
</tr>
<tr>
<td>CTCGA</td>
<td>1272535</td>
<td>1.2504804</td>
<td>5.1119194</td>
<td>9</td>
</tr>
<tr>
<td>CCTCG</td>
<td>1086970</td>
<td>1.2003711</td>
<td>5.1651883</td>
<td>8</td>
</tr>
<tr>
<td>AAAGA</td>
<td>1770400</td>
<td>1.1611295</td>
<td>5.138771</td>
<td>1</td>
</tr>
<tr>
<td>TGCGA</td>
<td>1193350</td>
<td>1.1499448</td>
<td>5.196163</td>
<td>3</td>
</tr>
<tr>
<td>TTGCG</td>
<td>990590</td>
<td>1.1324508</td>
<td>5.5174546</td>
<td>2</td>
</tr>
<tr>
<td>GATAT</td>
<td>1044855</td>
<td>0.96449</td>
<td>5.0834785</td>
<td>25</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
</body></html>